Error in TypeScript compilation: Cannot find name 'StdioOverrideFunction'

Hi, I'm trying to use your library in a project for work, but running `npm run build` in our project fails due to the following error:

```
node_modules/wasmagic/dist/index.d.ts:63:13 - error TS2304: Cannot find name 'StdioOverrideFunction'.

63     stdio?: StdioOverrideFunction;
               ~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/wasmagic/dist/index.d.ts:63
```

I see that the mentioned `StdioOverrideFunction` is defined in the `types` folder in this repo, but not in `src/index.ts` and it's apparent that the TypeScript compiler isn't including that type into the final `dist/index.d.ts`.

Can you fix this? It's breaking TypeScript builds at the moment.

--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@ dist/test/integration/foobar_magic dist/test/integration/png_magic dist/test/int
 
 package: dist/index.js dist/libmagic.LICENSE
 
-dist/index.js: $(ts_files) dist/libmagic-wrapper.js
+dist/index.js: $(ts_files) dist/libmagic-wrapper.js dist/LibmagicModule.d.ts dist/StdioOverrideFunction.d.ts
 	npx tsc -d
 
 dist/libmagic-wrapper.js: src/libmagic-wrapper.c dist/magic.mgc dist/libmagic.so dist/libmagic-wrapper.d.ts
@@ -31,8 +31,14 @@ dist/libmagic-wrapper.js: src/libmagic-wrapper.c dist/magic.mgc dist/libmagic.so
 	-o dist/libmagic-wrapper.js \
 	src/libmagic-wrapper.c
 
-dist/libmagic-wrapper.d.ts: types/libmagic-wrapper.d.ts
-	cp types/libmagic-wrapper.d.ts dist/libmagic-wrapper.d.ts
+dist/libmagic-wrapper.d.ts: src/libmagic-wrapper.d.ts
+	cp src/libmagic-wrapper.d.ts dist/libmagic-wrapper.d.ts
+
+dist/LibmagicModule.d.ts: src/LibmagicModule.d.ts
+	cp src/LibmagicModule.d.ts dist/LibmagicModule.d.ts
+
+dist/StdioOverrideFunction.d.ts: src/StdioOverrideFunction.d.ts
+	cp src/StdioOverrideFunction.d.ts dist/StdioOverrideFunction.d.ts
 
 dist/libmagic.LICENSE: vendor/file/COPYING
 	cp vendor/file/COPYING dist/libmagic.LICENSE

--- a/src/LibmagicModule.d.ts
+++ b/src/LibmagicModule.d.ts
@@ -1,7 +1,4 @@
-type StdioOverrideFunction = (
-  stdioName: "stdout" | "stderr",
-  text: string,
-) => void;
+import StdioOverrideFunction from "./StdioOverrideFunction";
 
 interface LibmagicModule extends EmscriptenModule {
   FS: typeof FS;
@@ -9,3 +6,5 @@ interface LibmagicModule extends EmscriptenModule {
   cwrap: typeof cwrap;
   printOverride: StdioOverrideFunction;
 }
+
+export default LibmagicModule;

--- a/src/StdioOverrideFunction.d.ts
+++ b/src/StdioOverrideFunction.d.ts
@@ -1,0 +1,6 @@
+type StdioOverrideFunction = (
+  stdioName: "stdout" | "stderr",
+  text: string,
+) => void;
+
+export default StdioOverrideFunction;

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,7 @@
-import libmagicFactory from "../dist/libmagic-wrapper";
+import libmagicFactory from "./libmagic-wrapper";
+import LibmagicModule from "./LibmagicModule";
+import StdioOverrideFunction from "./StdioOverrideFunction";
+export { default as StdioOverrideFunction } from "./StdioOverrideFunction";
 
 declare function wrappedDetect(bufPointer: number, bufLength: number): string;
 

--- a/src/libmagic-wrapper.d.ts
+++ b/src/libmagic-wrapper.d.ts
@@ -1,2 +1,4 @@
+import LibmagicModule from "./LibmagicModule";
+
 declare const factory: EmscriptenModuleFactory<LibmagicModule>;
 export default factory;

--- a/src/test/integration/index.ts
+++ b/src/test/integration/index.ts
@@ -2,7 +2,7 @@ import * as fs from "fs";
 import * as path from "path";
 import { glob } from "glob";
 import { cases } from "./data";
-import { WASMagic, WASMagicFlags } from "../../index";
+import { WASMagic, WASMagicFlags, StdioOverrideFunction } from "../../index";
 
 describe("WASMagic", () => {
   describe("Default magic file", () => {
@@ -49,6 +49,24 @@ describe("WASMagic", () => {
         expect(magicText.detect(fileBuf)).toBe(expectedText);
       },
     );
+
+    test("WASMagic throws when it fails to load", async () => {
+      let err: Error | undefined;
+      try {
+        await WASMagic.create({
+          loadDefaultMagicfile: false,
+          magicFiles: [
+            // To force a loading failure
+            Buffer.from("FOOOOOOOBAR"),
+          ],
+        });
+      } catch (_err) {
+        err = _err as Error;
+      }
+
+      expect(err).toBeInstanceOf(Error);
+      expect(err?.message.startsWith("WASMagic Load Error: ")).toBe(true);
+    });
   });
 
   describe("Specified foobar custom magic file, with default magic file", () => {
@@ -111,5 +129,38 @@ Some made up stuff
         expect(magic.detect(fs.readFileSync(file))).toBe("text/plain");
       },
     );
+  });
+
+  describe("StdioOverrideFunction", () => {
+    describe("StdioOverrideFunction fires when WASMagic fails to load", () => {
+      let stdioType: string | undefined;
+      let text: string | undefined;
+      const stdio: StdioOverrideFunction = (_stdioType, _text) => {
+        stdioType = _stdioType;
+        text = _text;
+      };
+
+      beforeAll(async () => {
+        try {
+          await WASMagic.create({
+            loadDefaultMagicfile: false,
+            magicFiles: [
+              // To force a loading failure
+              Buffer.from("FOOOOOOOBAR"),
+            ],
+            stdio,
+          });
+        } catch (_err) {
+          // Ignoring _err
+        }
+      });
+
+      test("stdioType is a String", () =>
+        expect(typeof stdioType).toBe("string"));
+      test("stdioType is stderr", () => expect(stdioType).toBe("stderr"));
+      test("text is a String", () => expect(typeof text).toBe("string"));
+      test("text is a known error", () =>
+        expect(text).toBe("0, 1: Warning: offset `FOOOOOOOBAR' invalid"));
+    });
   });
 });

